include/fs/littlefs.h

```diff
@@ -17,6 +17,16 @@
 extern "C" {
 #endif
 
+static inline bool littlefs_on_blkdev(struct fs_mount_t *mountp)
+{
+#ifdef CONFIG_FS_LITTLEFS_BLK_DEV
+	if (!strcmp(mountp->storage_dev, "SDMMC")) {
+		return true;
+	}
+#endif
+	return false;
+}
+
```

Comment:
I am wondering whether the `#ifdef` should go around the function and `#else` should be provided with
definition
```
#define littlefs_on_blkdev(ignore) false
```
The definition is only for the linker to not trigger warnings and calls like
```
if (littlefs_on_blkdev(sth)) {
```
Will optimize out easier as it will be immediately seen that this thing resolves at compile time to:
```
if (false) {
}
```
which means that the code is just not needed.
Note that inline is just a suggestion a compiler does not have to take.

Reply:
I can do as you advise here. No problem.

Reply:
"Will optimize out easier as it will be immediately seen that this thing resolves at compile time to:"

There will be no code size changes between these two techniques when the compiler has enabled optimizations.

It is true that inline is just a hint, but when the function is static the compiler will know of all usages and will optimize appropriately.

Reply:
Putting compiler vendor and how optimizations work aside, it will not hurt to help a compiler and show what is our intention. 
Here https://github.com/zephyrproject-rtos/zephyr/pull/40096/files is an example where addition of a few lines and change of ~2 results, unintentionally, in code reduction by tenths of bytes, on default Zephyr settings, which is counter intuitive.

Anyway, I will not oppose if the function is left there, if still needed, but with the `if` changed to something like:
```
if (IS_ENABLED(CONFIG_FS_LITTLEFS_BLK_DEV) && !strcmp(mountp->storage_dev, "SDMMC")) {
...
}
```
which looks better.